Go: Odd warning when compiling a Spin app for the first time

My guess is that this is totally harmless, but since it comes from within our own code:

```console
$ tinygo build -o favicon.wasm -target=wasi -wasm-abi=generic -gc=leaking main.go
/Users/technosophos/Code/Rust/spin/sdk/go/http/wasi-outbound-http.c:75:3: warning: switch condition has boolean value [-Wswitch-bool]
  switch (ptr->tag)
  ^       ~~~~~~~~
/Users/technosophos/Code/Rust/spin/sdk/go/http/wasi-outbound-http.c:93:3: warning: switch condition has boolean value [-Wswitch-bool]
  switch (ptr->tag)
  ^       ~~~~~~~~
2 warnings generated.
```

--- a/sdk/go/http/http_internals.go
+++ b/sdk/go/http/http_internals.go
@@ -6,7 +6,7 @@
 
 package http
 
-// #cgo CFLAGS: -Wall
+// #cgo CFLAGS: -Wno-unused-parameter -Wno-switch-bool
 // #include "wasi-outbound-http.h"
 // #include<stdlib.h>
 import "C"

